src/@types/common.ts

```diff
@@ -30,9 +30,9 @@ type Join<K, P> = K extends string | number ?
         `${K}${"" extends P ? "" : "."}${P}`
         : never : never;
 
-type Prev = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ...0[]];
+type Prev = [never, 0, 1, 2, 3, ...0[]];
 
-export type Leaves<T, D extends number = 5> = [D] extends [never] ? never : T extends object ?
+export type Leaves<T, D extends number = 3> = [D] extends [never] ? never : T extends object ?
```

Comment:
I'm not sure I understand what's going on here - either the change or the original code

Reply:
It tries to calculate all possible "leaf" values within an object in string dot notation format e.g. "user.name" for a `Room` object, the 5 -> 3 is the recursion limit, the typescript update got upset at the recursiveness of things

Reply:
:heavy_plus_sign: I would appreciate a comment here